### ISSUE TITLE
about the psc_frontend

### ISSUE THREAD

**Issue by Synlvejo** [readex-eu/readex-ptf#2]
Hello,
I used the following env and command:

export SCOREP_ENABLE_TRACING="false"
export SCOREP_ENABLE_PROFILING="true"
export SCOREP_PROFILING_FORMAT=cube_tuple

export SCOREP_METRIC_PLUGINS=x86_energy_sync_plugin
export SCOREP_METRIC_X86_ENERGY_SYNC_PLUGIN=*/E
export SCOREP_METRIC_PLUGINS_SEP=";"
export SCOREP_METRIC_X86_ENERGY_SYNC_PLUGIN_CONNECTION="INBAND"
export SCOREP_METRIC_X86_ENERGY_SYNC_PLUGIN_VERBOSE="WARN"
export SCOREP_METRIC_X86_ENERGY_SYNC_PLUGIN_STATS_TIMEOUT_MS=1000

export SCOREP_SUBSTRATE_PLUGINS=rrl
export SCOREP_RRL_PLUGINS=cpu_freq_plugin,uncore_freq_plugin
export SCOREP_RRL_VERBOSE="WARN"

psc_frontend --apprun="./mg.C.x" 
                      --ompnumthreads=2 
                      --phase=main 
                      --tune=readex_tuning 
                      --config-file=readex_config.xml

But there is not command output or .json file，the only output is a scorep folder and command message as follow:

Periscope Performance Analysis Tool (ver. 2.0)
[psc_frontend][INFO:fe] Preparing to start the performance tuning...
[psc_frontend][INFO:fe] Starting application ./mg.C.x using 1 MPI procs and 2 OpenMP threads...
Score-P RRL plugin:[localhost.localdomain][P:234480][R:-1][DEBUG][]: [UTIL] Access to environment variable 'SCOREP_RRL_VERBOSE'
[cpu_freq]CPU_FREQ tuning plugin: initializing
[UNCORE_FREQ_TP]UNCORE_FREQ tuning plugin: initializing
<there is the message of mg benchmark >

So I want to know if my env was wrong? Or I missed some keywords?

Thanks for any reply.

**Comment by madhurakswamy** [comment]
Hello,
To get the config file, you first have to run the application as you normally would without PTF:
srun -n x ./mg.C.x

You can then run PTF using this config file as input using the command  psc_frontend --apprun="./mg.C.x" ... as you already did.

**Comment by Synlvejo** [comment]
> Hello,
> To get the config file, you first have to run the application as you normally would without PTF:
> srun -n x ./mg.C.x
> You can then run PTF using this config file as input using the command  psc_frontend --apprun="./mg.C.x" ... as you already did.

Thanks.
Maybe I did't say what I mean.
I had got the readex-config.xml , and in the next step to use PTF(psc_frontend) to get the .json file and scenaries energy.
The resion I used "main" as my phase name is that I did not macro the LOOP manually.

**Comment by madhurakswamy** [comment]
Hello,
I think that it is because you have not enabled the output from PTF using --info=2000 flag in the psc_frontend command. This outputs all the information. If you want less info, please use a number between 2 and 10. Also, please comment this line:
export SCOREP_PROFILING_FORMAT=cube_tuple
It is not needed for the PTF run, and causes unnecessary overhead.

**Comment by Synlvejo** [comment]
> Hello,
> I think that it is because you have not enabled the output from PTF using --info=2000 flag in the psc_frontend command. This outputs all the information. If you want less info, please use a number between 2 and 10. Also, please comment this line:
> export SCOREP_PROFILING_FORMAT=cube_tuple
> It is not needed for the PTF run, and causes unnecessary overhead.

Thanks ,
I add the --info=2000 and export SCOREP_PROFILING_FORMAT=cube_tuple.There are lots of :

[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
[psc_frontend][DBG:1:fe] components started up: 0 (of 1)

And end with Aborted.

I'm in the PDTA(3) in the video "READEX AlphaPrototype Video Tutorial". And I can't get the message like Optium Scenario , All Result, Best configurations and tuning_model.json .
But I don't know why .
Maybe there are other resons?

**Comment by madhurakswamy** [comment]
Please do not export cube_tuple. As i said, it will cause unnecessary overhead. I can recommend the following:
1) use the --force-localhost flag with the psc_frontend command and see if output is generated.
2) set --mpinumprocs=1 in the psc_frontend command when using one process

Also, could you paste the contents of your ~/.periscope file? The configuration of PTF is loaded from the .periscope configuration file. It has to be located in your home directory for PTF to work. If you don't have this file, please copy readex-ptf/templates/periscope.sample into .periscope under your home directory.

**Comment by Synlvejo** [comment]
> Please do not export cube_tuple. As i said, it will cause unnecessary overhead. I can recommend the following:
> 
> use the --force-localhost flag with the psc_frontend command and see if output is generated.
> set --mpinumprocs=1 in the psc_frontend command when using one process
> 
> Also, could you paste the contents of your ~/.periscope file? The configuration of PTF is loaded from the .periscope configuration file. It has to be located in your home directory for PTF to work. If you don't have this file, please copy readex-ptf/templates/periscope.sample into .periscope under your home directory.

Hello,
The ~/.perscope:
MACHINE = Periscope
SITE = NONE
REGSERVICE_HOST = localhost.localdomain
REGSERVICE_PORT = 50041
REGSERVICE_HOST_INIT = localhost
REGSERVICE_PORT_INIT = 50001
APPL_BASEPORT = 51000
AGENT_BASEPORT = 50002

And I use this example file to overwrite the ~/.perscope: 
MACHINE = localhost
SITE = LiveDVD
REGSERVICE_HOST = localhost.localdomain
REGSERVICE_PORT = 50459
REGSERVICE_HOST_INIT = localhost
REGSERVICE_PORT_INIT = 50001
APPL_BASEPORT = 51000
AGENT_BASEPORT = 50002

Run this command without export cube_tuple: 
psc_frontend --apprun="./mg.C.x" --mpinumprocs=1 --ompnumthreads=2 --phase=main --tune=readex_tuning --config-file=readex_config.xml --info=2 --force-localhost --selective-info=AutotuneAll,AutotunePlugins
The result:

[psc_frontend][INFO:fe] Preparing to start the performance tuning...
[psc_frontend][DBG:2:fe] RegistryService(): Started.
[psc_frontend][DBG:2:fe] Localhost startup activated! Will NOT use SSH for starting the agents!
[psc_frontend][DBG:1:fe] Using phase region "main" 
[psc_frontend][DBG:2:fe] Frontend up and running on localhost.localdomain:50002
[psc_frontend][DBG:2:fe] Using maxfan=4
[psc_frontend][DBG:2:fe] Using output file: properties_tune_76053.psc
[psc_frontend][DBG:2:fe] Contacting registry service...
[psc_frontend][DBG:2:fe] Registration successful!
[psc_frontend][INFO:fe] Starting application ./mg.C.x using 1 MPI procs and 2 OpenMP threads...
[psc_frontend][DBG:1:fe] Starting application in interactive partition: >export SCOREP_ONLINEACCESS_ENABLE=true; export SCOREP_ENABLE_TRACING=false; export SCOREP_ONLINEACCESS_REG_PORT=50459; export SCOREP_ONLINEACCESS_REG_HOST=localhost.localdomain; export SCOREP_ONLINEACCESS_APPL_NAME=appl76053; export SCOREP_TUNING_PLUGINS_SEP=,; export SCOREP_TUNING_PLUGINS=OpenMPTP; export SCOREP_TIMER="clock_gettime"; export OMP_NUM_THREADS=2;  mpiexec -n 1 ./mg.C.x<
Score-P RRL plugin:[localhost.localdomain][P:76064][R:-1][DEBUG][]: [UTIL] Access to environment variable 'SCOREP_RRL_VERBOSE'
[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
[cpu_freq]CPU_FREQ tuning plugin: initializing
[UNCORE_FREQ_TP]UNCORE_FREQ tuning plugin: initializing

 NAS Parallel Benchmarks (NPB3.4-OMP) - MG Benchmark

 No input file. Using compiled defaults 
[psc_frontend][DBG:1:fe] components started up: 0 (of 1)

 Size:  512x 512x 512  (class C)
 Iterations:                     20
 Number of available threads:     2

[psc_frontend][DBG:1:fe] components started up: 0 (of 1) x n
 Initialization time:          68.640 seconds

  iter   1
[psc_frontend][DBG:1:fe] components started up: 0 (of 1) x n
  iter   5
[psc_frontend][DBG:1:fe] components started up: 0 (of 1) x n
  iter  10
[psc_frontend][DBG:1:fe] components started up: 0 (of 1) x n

  iter  15
[psc_frontend][DBG:1:fe] components started up: 0 (of 1) x n
  iter  20
[psc_frontend][DBG:1:fe] components started up: 0 (of 1) x n

 Benchmark completed 
 VERIFICATION SUCCESSFUL 
 L2 Norm is  0.5706732285730E-06
 Error is    0.1665721602942E-11


 MG Benchmark Completed.
 Class           =                        C
 Size            =            512x 512x 512
 Iterations      =                       20
 Time in seconds =                    29.97
 Total threads   =                        2
 Avail threads   =                        2
 Mop/s total     =                  5194.63
 Mop/s/thread    =                  2597.32
 Operation type  =           floating point
 Verification    =               SUCCESSFUL
 Version         =                      3.4
 Compile date    =              29 Apr 2020

 Compile options:
    F77          = scorep gfortran
    FLINK        = $(FC)
    F_LIB        = (none)
    F_INC        = (none)
    FFLAGS       = -O3 -fopenmp
    FLINKFLAGS   = $(FFLAGS)
    RAND         = randi8

 Please send all errors/feedbacks to:

 NPB Development Team
 npb@nas.nasa.gov

[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
Aborted

Looks like no more effect
Thanks for reply.

**Comment by madhurakswamy** [comment]
I just saw that you are using --tune=readex_tuning, which is outdated, and only works for the old version of PTF.  Please use --tune=readex intraphase, and also refer the user guide instead of the video. The user guide has been updated for the newest version.

**Comment by Synlvejo** [comment]
> I just saw that you are using --tune=readex_tuning, which is outdated, and only works for the old version of PTF.  Please use --tune=readex intraphase, and also refer the user guide instead of the video. The user guide has been updated for the newest version.

Hello，
I'm coming back~

**I use the env:**

export SCOREP_METRIC_PLUGINS=x86_energy_sync_plugin
export SCOREP_METRIC_PLUGINS_SEP=";"
export SCOREP_METRIC_X86_ENERGY_SYNC_PLUGIN="BLADE/E"
export SCOREP_METRIC_X86_ENERGY_PLUGIN_INTERVALL_US=100000
export SCOREP_METRIC_X86_ENERGY_SYNC_PLUGIN_OFFSET=70
export SCOREP_SUBSTRATE_PLUGINS=rrl
export SCOREP_RRL_PLUGINS=cpu_freq_plugin,uncore_freq_plugin

**And use the command:**

psc_frontend --apprun="./XSBench -t 28" --mpinumprocs=1 --ompnumthreads=28 --phase=main --config-file=/home/xzb/denchmark/XSBench/src1/XSBench1/readex_config.xml --tune=readex_intraphase --info=2000 --force-localhost

**And get the command result like this with only once execution without .json file:(the same problem is above all benchmarks)**

Calcula[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
Calcu[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
Calculating XS's... (49% complete[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
Calculating XS's... (73% [psc_frontend][DBG:1:fe] components started up: 0 (of 1)
Calculating XS's.[psc_frontend][DBG:7:fe] PeerAcceptor::handle_close
[psc_frontend][DBG:1:fe] components started up: 0 (of 1)
Aborted (core dumped)

**Then I try to use gdb to find what's wrong with it , I get this :**

Program received signal SIGABRT, Aborted.
0x00007ffff5aa2387 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007ffff5aa2387 in raise () from /lib64/libc.so.6
#1  0x00007ffff5aa3a78 in abort () from /lib64/libc.so.6
#2  0x0000000000538fbe in ApplicationStarter::startApplication (this=0x7ffff7f16010, 
    isFirstStart=isFirstStart@entry=true) at ../frontend/src/ApplicationStarter.cc:99
#3  0x000000000053910a in ApplicationStarter::runApplication (this=this@entry=0x7ffff7f16010)
    at ../frontend/src/ApplicationStarter.cc:163
#4  0x00000000004d9389 in application_then_agents_start (fe=<optimized out>, starter=0x7ffff7f16010)
    at ../frontend/src/frontend_statemachine.cc:161
#5  0x00000000004dbc75 in frontend_statemachine::autotune_msm::start_application_and_agent_network (
    this=this@entry=0x7fffffffd240, reactor=...) at ../frontend/src/frontend_statemachine.cc:636
#6  0x00000000004dda8d in frontend_statemachine::autotune_msm::initialize_plugin_action (
    this=this@entry=0x7fffffffd240, evt=...) at ../frontend/src/frontend_statemachine.cc:189
#7  0x00000000004d06d9 in boost::msm::front::state_machine_def<frontend_statemachine::autotune_msm, boost::msm::front::default_base_state>::a_row<frontend_statemachine::autotune_msm::InitialState, frontend_statemachine::initialize_plugin, frontend_statemachine::autotune_msm::InitializingPlugin, &frontend_statemachine::autotune_msm::initialize_plugin_action>::action_call<boost::msm::back::state_machine<frontend_statemachine::autotune_msm, boost::parameter::void_, boost::parameter::void_, boost::parameter::void_, boost::parameter::void_>, frontend_statemachine::autotune_msm::InitialState, frontend_statemachine::autotune_msm::InitializingPlugi……

**Besides, there is part of my readex-config.xml:**

```
<dynamism>
            <dyn_time_variation/>
            <dyn_comIn_variation/>
        </dynamism>
    </readex-dyn-detect>
    <tuningParameter>
        <frequency>
            <min_freq/>1200<min_freq/>
            <max_freq/>2200<max_freq/>
            <freq_step/>400<freq_step/>
            <default/>2200<default/>
        </frequency>
        <uncore>
            <min_freq/>1200<min_freq/>
            <max_freq/>2400<max_freq/>
            <freq_step/>600<freq_step/>
            <default/>2400<default/>
        </uncore>
        <openMPThreads>
            <lower_value/>4<lower_value/>
            <step/>12<step/>
        </openMPThreads>
    </tuningParameter>
    <objectives>
        <objective>Energy</objective>
        <objective>Time</objective>
        <objective>EDP</objective>
        <objective>ED2P</objective>
        <objective>CPUEnergy</objective>
        <objective>TCO</objective>
    </objectives>
    <CostPerJoule>8.0000000000000002e-08</CostPerJoule>
    <CostPerCoreHour>1</CostPerCoreHour>
    <periscope>
        <metricPlugin>
            <name>x86_energy_sync_plugin</name>
        </metricPlugin>
        <metrics>
            <node_energy>x86_energy/BLADE/E</node_energy>
        </metrics>
        <searchAlgorithm>
            <name>exhaustive</name>
        </searchAlgorithm>
        <atp>
            <searchAlgorithm>
                <name>exhaustive_atp</name>
            </searchAlgorithm>
        </atp>
        <tuningModel>
            <file_path/>./tuning_model.json<file_path/>
        </tuningModel>
    </periscope>
    <scorep>
        <tuningSubstrate/>rrl<tuningSubstrate/>
    </scorep>
</Configuration>
```

**So is my psc_fronted tool wrong in building? Can you find some useful message from this?**

What's more, the cpu_freq_plugins and uncore_freq_plugin was initializing successfully but I can't see change of my cf and ucf,TAT

Thanks for any responds.

**Comment by Synlvejo** [comment]
> I just saw that you are using --tune=readex_tuning, which is outdated, and only works for the old version of PTF. Please use --tune=readex intraphase, and also refer the user guide instead of the video. The user guide has been updated for the newest version.

Hello,

I tried to rebuild PTF by using:

'''
../configure --prefix=/home/xzb/PTF-2 --enable-developer-mode --with-starter=interactive --with-ace-include=/home/xzb/ACE_wrappers/lib --with-ace-lib=/home/xzb/ACE_wrappers/include --with-boost-include=/home/xzb/boost/include --with-boost-lib=/home/xzb/boost/lib --with-cube-include=/home/xzb/Scorep/include --with-cube-include=/home/xzb/Scorep/lib --with-scorep-include=/home/xzb/Scorep/include --with-scorep-lib=/home/xzb/Scorep/lib --with-cereal-include=/usr/include/cereal
'''

And got this error:

```
  CXX      hagent/src/psc_hlagent-hagent_accl_handler.o
  CXX      hagent/src/psc_hlagent-hagent_accl_statemachine.o
  CXX      readex/cube_tools/tuning_potential/src/readex_dyn_detect-AdjacencyMatrix.o
  CXX      readex/cube_tools/tuning_potential/src/readex_dyn_detect-DetectRegion.o
../readex/cube_tools/tuning_potential/src/DetectRegion.cc: In function ‘int printSummary(PhaseRegion*, const std::list<SignificantRegion*>&, double, double, double, std::vector<std::basic_string<char> >&, std::vector<std::basic_string<char> >&, bool, const string&, double)’:
../readex/cube_tools/tuning_potential/src/DetectRegion.cc:723:29: error: ‘next’ is not a member of ‘boost’
                 if ( boost::next( region ) == com_intensity_variation_reg.end() )
                             ^~~~
../readex/cube_tools/tuning_potential/src/DetectRegion.cc:723:29: note: suggested alternative: ‘get’
                 if ( boost::next( region ) == com_intensity_variation_reg.end() )
                             ^~~~
                             get
  CXX      readex/cube_tools/tuning_potential/src/readex_dyn_detect-TuningPotential.o
  CXX      readex/cube_tools/tuning_potential/src/readex_dyn_detect-ProcessingIntensity.o
make[2]: *** [Makefile:9783: readex/cube_tools/tuning_potential/src/readex_dyn_detect-DetectRegion.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/xzb/PTF/build'
make[1]: *** [Makefile:10065: all-recursive] Error 1
make[1]: Leaving directory '/home/xzb/PTF/build'
make: *** [Makefile:3021: all] Error 2

```
boost version 1.67
